### PR TITLE
Initialized repository from template

### PR DESCRIPTION
**Pull request by josephlewis42**
Initialized repository from the Python bootstrap template here: https://github.com/openzim/_python-bootstrap.

Part of #1

### CODE REVIEW

**Review by benoit74 — CHANGES_REQUESTED** [review]
Very good, few things to fix but already mostly OK, thank you!

A scraper is supposed to support only one Python version (3.12 as of now) since it is not meant to be reused in a "bigger soft" (like a library would). Supporting multiple Python version is hence unneeded additional work, since it is very likely it will be useless. 

You also need at some point to update the dependencies used to latest versions (in pyproject.toml, but also in .pre-commit-config.yaml). Doing this in every PR is a pain and unneeded, but since this is the initial one I suggest you update dependencies to latest versions now, and we will do it once again just before releasing the first version of this scraper.

Regarding last point, you should definitely have a look at https://github.com/openzim/_python-bootstrap/pull/45

**Comment by josephlewis42** [comment]
> You also need at some point to update the dependencies used to latest versions (in pyproject.toml, but also in .pre-commit-config.yaml). Doing this in every PR is a pain and unneeded, but since this is the initial one I suggest you update dependencies to latest versions now, and we will do it once again just before releasing the first version of this scraper.

Does dependabot work with how these repos are set up or is there a strong preference for using specifically tagged versions?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM python:3.11-slim-bookworm
```
[the remark this answers is not in the document]
Comment (benoit74) [reply Dockerfile]: Looks like I forgot this one in my PR from yesterday, yes please open a PR with this.

**Comment by kelson42** [comment]
I would recommend to make a basic README file similar to in other repos.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+[project]
+name = "devdocs2zim"
+requires-python = ">=3.12,<3.13"
+description = "Make ZIM files from DevDocs.io"
+readme = "README.md"
+dependencies = [
+    "requests==2.31.0",
+]
+dynamic = ["authors", "classifiers", "keywords", "license", "version", "urls"]
+
```
Comment (benoit74) [inline pyproject.toml:14]: I just realized that you miss a section to configure and trigger the automatic population of authors, classifiers, keywords, license and urls metadata with hatch-openzim plugin.

Following configuration is probably OK to populate default metadata for scrapers + add a "devdocs" keyword.
```
[tool.hatch.metadata.hooks.openzim-metadata]
kind = "scraper"
additional-keywords = ["devdocs"]
```

At least this is the "usual" configuration on other scrapers, see e.g.:
- https://github.com/openzim/ted/blob/main/pyproject.toml#L23-L25
- https://github.com/openzim/youtube/blob/main/scraper/pyproject.toml#L27-L29

**Review by benoit74 — CHANGES_REQUESTED** [review]
LGTM, only one last thing I failed to find in first review, sorry for that.

Since otherwise everything is OK, please directly do the modification and rewrite the commit history of your branch to only keep meaningful commits (i.e. squash / fixup commits). The goal is to remove commits like "Fixed PR comments". These commits are more linked to the review process and won't bring much value / make maintenance harder in one or two years ... This will obviously require a force push on your branch, but this is OK once review is over (or mostly over) - but avoid to do this before the end of the review, since otherwise it makes the second review harder.

**Comment by benoit74** [comment]
> I would recommend to make a basic README file similar to in other repos.

You mean, something more targeted towards usage by everyone rather than how to develop? I considered this would come in another PR once the software has settled, I don't think we already know how the software will really work / which parameters will be necessary. But yes, this is going to be needed.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (josephlewis42) [reply Dockerfile]: Created: https://github.com/openzim/_python-bootstrap/pull/46

**Comment by josephlewis42** [comment]
> This will obviously require a force push on your branch, but this is OK once review is over (or mostly over) - but avoid to do this before the end of the review, since otherwise it makes the second review harder.

Done! I didn't realize that "Squash and Merge" was disabled on this repo.

> I would recommend to make a basic README file similar to in other repos.

I added a bit more here with some example flags and installation instructions. There's still the warning that they're for example purposes only.

**Review by benoit74 — APPROVED** [review]
LGTM, thank you !

**Pull request merged**